zoomable-image/coil/src/androidTest/kotlin/me/saket/telephoto/zoomable/coil/CoilImageSourceTest.kt

```diff
@@ -249,6 +256,67 @@ class CoilImageSourceTest {
     }
   }
 
+  @Test fun uses_updated_async_placeholder_size_when_available() = runTest {
+    val imageUrl = serverRule.server.url("/slow_placeholder_image.png")
+    lateinit var state: ZoomableImageState
+    var displayImage by mutableStateOf(false)
+    var placeholderLoaded = false
+
+    rule.setContent {
+      state = rememberZoomableImageState()
+
+      val placeholderPainter = rememberAsyncImagePainter(
+        model = ImageRequest.Builder(context)
+          .data(imageUrl)
+          .allowHardware(false) // Unsupported by Screenshot.capture()
+          .listener(
+            onSuccess = { _, _ ->
+              placeholderLoaded = true
+            }
+          )
+          .build()
+      )
+      val source = object : ZoomableImageSource {
```

Comment:
I'm slightly surprised this does not need to be remembered.

Reply:
Hmm, I suppose that it's not storing any state internally that requires remembering, so even if it gets recreated on recomposition it is still doing the right thing?

Reply:
Ah right, I often forget that the compose compiler only looks at functions. From its perspective, `resolve()` does not require a new composition branch even if the `ZoomableImageSource` instance is updated.